### ISSUE TITLE
Mobile navigation tab order does not follow visual order

### ISSUE THREAD

**Issue by aduth** [uswds/uswds#3526]
## Description

When tabbing through elements in the mobile menu navigation drawer, the tab order may not align to an expected order based on what is visually present on the page. The search field is shown visually toward the top of the drawer, but is last in the tab order. This can cause confusion for users who navigate using keyboard tabbing, or iOS VoiceOver / Android TalkBack gesture navigation.

## Steps to reproduce the issue

1. Navigate to [Basic Header demo](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/release-2.7.1/components/preview/header--basic.html)
2. Shrink the page, or zoom in to the point where the mobile navigation styling is activated
3. <kbd>Tab</kbd> to and activate the menu button using <kbd>Space</kbd>
4. Observe that focus is placed on the "Close" button in the navigation drawer
5. Press <kbd>Tab</kbd>

**Expected behavior:** Focus is placed in the search field, as it is visually located immediately following the current active element
**Actual behavior:** Focus is placed in the navigation menu links. Continuing to press <kbd>Tab</kbd>, you will observe that the search field is last in the tab order

## Additional information

Screen recording:

![uswds-menu-tab-order](https://user-images.githubusercontent.com/1779930/85897377-30e6dd80-b7c8-11ea-86a7-f66249e7d833.gif)

Relevant success criteria [2.4.3 Focus Order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html):

>For example, a screen reader user interacts with the programmatically determined reading order, while a sighted keyboard user interacts with the visual presentation of the Web page. Care should be taken so that the focus order makes sense to both of these sets of users and does not appear to either of them to jump around randomly.

Relevant code, which places the navigation _visually_ last, even if in markup it occurs before the search field:

https://github.com/uswds/uswds/blob/9ca144f77120b18e6f22544fe66f7e3f727b4e4e/src/stylesheets/components/_navigation.scss#L73

Relevant ["Accessibility concerns" guidance for `order` CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/order#Accessibility_concerns):

>Using the `order` property will create a disconnect between the visual presentation of content and DOM order. This will adversely affect users experiencing low vision navigating with the aid of assistive technology such as a screen reader. If the visual (css) order is important, then screen reader users will not have access to the correct reading order.

Note: This issue does not occur on the main [USWDS website](https://designsystem.digital.gov/), only because the `.usa-nav__primary` on which the `order` styling is applied is wrapped by a `.usa-nav__inner`.

Possible remediation options:

- Search should be visually displayed the same as it is in markup order (i.e. remove `order`, allow search field to show at the bottom of the drawer)
- Search should be moved before the navigation links in the markup. Note that this could have adverse effects on the intended desktop styling.
- Search field should be programmatically moved to occur in the DOM prior to navigation links at smaller viewport sizes. Note that it's unclear if assistive technologies is accommodating to these programmatic changes.

**Comment by mejiaj** [comment]
Related to #3920.
